cmd/manager/main.go

```diff
@@ -537,6 +538,13 @@ func asyncTasks(mgr manager.Manager, cfg *rest.Config, managedNamespaces []strin
 		r.Start(licensing.ResourceReporterFrequency)
 	}()
 
+	// Start the telemetry reporter
+	go func() {
+		mgr.GetCache().WaitForCacheSync(nil) // wait until k8s client cache is initialized
```

Comment:
Curious if we could do the cache sync in front of both go blocks (telemetry and resource reporter). Also, even if not introduced here, I wonder why we are ignoring the return value of `WaitForCacheSync`. It returns `false` if it was not able to sync the cache.

Reply:
I left just one above, but I'm not sure what is the proper behavior if it keeps returning `false`. Crash the operator?

Reply:
Good point and probably the explanation why the code is as it is right now.